src/formatting/syntux/session.rs

```diff
@@ -189,6 +189,15 @@ impl ParseSess {
         }
     }
 
+    pub(crate) fn line_bounds(&self, pos: BytePos) -> (BytePos, BytePos) {
+        let line = self.parse_sess.source_map().lookup_line(pos).ok();
+
+        match line {
+            Some(line_info) => line_info.sf.line_bounds(line_info.line),
+            None => (BytePos::from_usize(0), BytePos::from_usize(0)),
+        }
+    }
```

Comment:
Hmmm, I don't think we'll want to silently map a source map lookup error to hi/lo `0` like this. It doesn't really seem a likely condition but I have a nagging suspicion it could eventually be hit which would cause some weird formatting behavior. 

Probably best to set the return type to an option containing tuple, and then call explicitly unwrap from the caller to panic and fail loudly in that hypothetical case